src/test/run-pass/specialization/defaultimpl/specialization-trait-item-not-implemented.rs

```diff
@@ -8,30 +8,27 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// Test that non-method associated functions can be specialized
-
 #![feature(specialization)]
 
 trait Foo {
-    fn mk() -> Self;
+    fn foo_one(&self) -> &'static str;
+    fn foo_two(&self) -> &'static str;
 }
 
-default impl<T: Default> Foo for T {
-    fn mk() -> T {
-        T::default()
+struct MyStruct;
+
+default impl<T> Foo for T {
+    fn foo_one(&self) -> &'static str {
+        "generic"
```

Comment:
Maybe nice to have this invoke a `foo_three` that the regular impl supplies, just to complete the "calling back and forth" cycle.